Tests/ForEachAsyncTests.cs

```diff
@@ -31,6 +31,50 @@ public void SimpleSyncForEach()
 #pragma warning restore CS0612 // Type or member is obsolete
         }
 
+        [Test]
+        public async Task SimpleAsyncForEachWithSyncBreak()
+        {
+            IAsyncEnumerable<int> enumerable = new AsyncEnumerable<int>(
+                async yield =>
+                {
+                    for (int i = 0; i < 5; i++)
+                        await yield.ReturnAsync(i);
+                });
+
+            int counter = 0;
+            await enumerable.ForEachAsync(
+                number =>
+                {
+                    Assert.AreEqual(counter, number);
+                    counter++;
+                    if (counter == 3) enumerable.Break();
```

Comment:
I think this quick solution is a good compromise compared to introducing a loop state argument with the loop control. However, the `Break` extension method is available outside of the loop, and technically should not belong to the enumerable (nor the enumerator).
I suggest to introduce just a simple static class e.g. 'ForEachAsync' with the method `Break`:
```
#if !NETCOREAPP3_0
public static class ForEachAsync
{
    public static void Break() => throw new ForEachAsyncCanceledException();
}
```

So the usage would look like this:
```
await enumerable.ForEachAsync(
item =>
{
    if (...)
        ForEachAsync.Break();
});
```

Thoughts?